src/libs/cropOrRotateImage/index.ts

```diff
@@ -1,127 +1,19 @@
-
-    width = clamp(width, canvas.width);
-    height = clamp(height, canvas.height);
-    originX = clamp(originX, canvas.width);
-    originY = clamp(originY, canvas.height);
-
-    width = Math.min(originX + width, canvas.width) - originX;
-    height = Math.min(originY + height, canvas.height) - originY;
-
-    const result = document.createElement('canvas');
-    result.width = width;
-    result.height = height;
-
-    const context = result.getContext('2d');
-    context?.drawImage(canvas, originX, originY, width, height, 0, 0, width, height);
-
-    return result;
-}
-
-function convertCanvasToFile(canvas: HTMLCanvasElement, options: CropOrRotateImageOptions): Promise<File> {
-    return new Promise((resolve) => {
-        canvas.toBlob((blob) => {
-            if (!blob) {
-                return;
-            }
-            const file = new File([blob], options.name || 'fileName.jpeg', {type: options.type || 'image/jpeg'});
-            file.uri = URL.createObjectURL(file);
-            resolve(file);
-        });
-    });
-}
-
-/**
- * Loads image from specified url
- */
-function loadImageAsync(uri: string): Promise<HTMLCanvasElement> {
-    return new Promise((resolve, reject) => {
-        const imageSource = new Image();
-        imageSource.crossOrigin = 'anonymous';
-        const canvas = document.createElement('canvas');
-        imageSource.onload = () => {
-            canvas.width = imageSource.naturalWidth;
-            canvas.height = imageSource.naturalHeight;
-
-            const context = canvas.getContext('2d');
-            context?.drawImage(imageSource, 0, 0, imageSource.naturalWidth, imageSource.naturalHeight);
-
-            resolve(canvas);
-        };
-        imageSource.onerror = () => reject(canvas);
-        imageSource.src = uri;
-    });
-}
-
-/**
- * Crops and rotates the image on web
- */
+import {manipulateAsync} from 'expo-image-manipulator';
+import type {CropOrRotateImage} from './types';
+import getSaveFormat from './utils';
 
 const cropOrRotateImage: CropOrRotateImage = (uri, actions, options) =>
-    loadImageAsync(uri).then((originalCanvas) => {
-        const resultCanvas = actions.reduce((canvas, action) => {
-            if (action.crop) {
-                return cropCanvas(canvas, action.crop);
-            }
-            if (action.rotate) {
-                return rotateCanvas(canvas, action.rotate);
-            }
-            return canvas;
-        }, originalCanvas);
-        return convertCanvasToFile(resultCanvas, options);
+    new Promise((resolve) => {
+        const format = getSaveFormat(options.type);
+        manipulateAsync(uri, actions, {compress: options.compress, format}).then((result) => {
+            fetch(result.uri)
+                .then((res) => res.blob())
+                .then((blob) => {
+                    const file = new File([blob], options.name || 'fileName.jpeg', {type: options.type || 'image/jpeg'});
```

Comment:
I am wondering in which case will `options.name` will not be set and we'll have to fallback for `fileName.jpeg`? Can we use any other fallback? instead of hardcoding?